scripts/ci/validate-action-pinning.sh

```
@@ -1,36 +1,102 @@
 
 set -euo pipefail
 
-echo "Checking for non-SHA pinned actions..."
+STRICT_MODE=false
+CHECK_COMMITS=false
+
+# Parse arguments
+while [[ $# -gt 0 ]]; do
+  case "$1" in
+    --strict)
+      STRICT_MODE=true
```

Shellcheck (0.11.0)

[warning] STRICT_MODE appears unused. Verify use (or export if used externally).

(SC2034)